blueprint/core/blueprint_CanvasView.h

```diff
@@ -11,19 +11,411 @@
+        void init()
+        {
+            graphics = std::make_unique<juce::Graphics>(image);
+        }
+
+    private:
+        void registerNativeProperties()
+        {
+            setProperty("__setFillStyle", juce::var::NativeFunction {
+                    [=](const juce::var::NativeFunctionArgs& args) -> juce::var {
+                        jassert(graphics);
+                        jassert(args.numArguments == 1);
+
+                        //TODO: Implement fillStyle pattern
+                        //TODO: Implement fillStyle gradient
+
+                        const auto colourHex = args.arguments[0].toString();
+                        properties.fillStyle.setColour(juce::Colour::fromString(colourHex));
+
+                        graphics->setFillType(properties.fillStyle);
+
+                        return juce::var();
+                    }
+            });
+
+            setProperty("__setStrokeStyle", juce::var::NativeFunction {
+                    [=](const juce::var::NativeFunctionArgs& args) -> juce::var {
+                        jassert(graphics);
+                        jassert(args.numArguments == 1);
+
+                        //TODO: Implement strokeStyle pattern
+                        //TODO: Implement stokeStyle gradient
+
+                        const auto colourHex = args.arguments[0].toString();
+                        properties.strokeStyle.setColour(juce::Colour::fromString(colourHex));
+
+                        return juce::var();
+                    }
+            });
+
+            setProperty("__setLineWidth", juce::var::NativeFunction {
+                    [=](const juce::var::NativeFunctionArgs& args) -> juce::var {
+                        jassert(graphics);
+                        jassert(args.numArguments == 1);
+
+                        const int lineWidth = args.arguments[0];
+                        properties.lineWidth = lineWidth;
+
+                        return juce::var();
+                    }
+            });
+        }
+
+        void registerNativeFunctions()
+        {
+//            setProperty("rect", juce::var::NativeFunction {
+//                    [=](const juce::var::NativeFunctionArgs& args) -> juce::var {
+//                        jassert(args.numArguments == 4);
+//
+//                        const float x = args.arguments[0];
+//                        const float y = args.arguments[1];
+//                        const float width = args.arguments[2];
+//                        const float height = args.arguments[3];
+//
+//                        // TODO: For some reason this operation closes the current path.
+//                        //       The API docs suggest addRectangle should be added as a new sub-path
+//                        //       leaving the previous path open.
+//                        //       Fix.
+//                        path.addRectangle(x, y, width, height);
```

Comment:
:+1: we can manually do some `lineTo`'s here I think, right?

Reply:
I tried a move to before the addRectangle call bit it appeared to close the previous path. If you'd prefer I can just remove the commented out block of code and fix this in a commit after this. I don't think rect() is essential at this point.

Reply:
Hmm. No worries, let's leave the commented block in. I think having the bones in place there will be helpful for our own remembering, and helpful if we have a new contributor who wants a good, small first task